File: apps/web/src/lib/validators/TextGeneration.ts
import { z } from "zod";

export const causalModelValidator = z.enum([
    // ChatGPT
    "gpt-3.5-turbo",
    // GPT-3
    "text-davinci-003",
    "text-curie-001",
    "text-babbage-001",
    "text-ada-001",
    // Codex
    "code-davinci-002",
    "code-cushman-001"
]);
export type CausalModel = z.infer<typeof causalModelValidator>;

export const causalGenerationValidator = z.object({
    text: z.string()
});
export type CausalGeneration = z.infer<typeof causalGenerationValidator>;

export const streamingCausalGenerationValidator = z.object({
    text: z.string()
});
export type StreamingCausalGeneration = z.infer<typeof streamingCausalGenerationValidator>;

File: apps/api/src/solidchain/schemas/text_generation.py
from enum import Enum
from typing import Any, List

from pydantic import BaseModel


class CausalModel(str, Enum):
    GPT_3_5_TURBO = "gpt-3.5-turbo"
    TEXT_DAVINCI_003 = "text-davinci-003"
    TEXT_CURIE_001 = "text-curie-001"
    TEXT_BABBAGE_001 = "text-babbage-001"
    TEXT_ADA_001 = "text-ada-001"
    CODE_DAVINCI_002 = "code-davinci-002"
    CODE_CUSHMAN_001 = "code-cushman-001"


class CausalGeneration(BaseModel):
    text: str
    steps: List[Any] = []


class StreamingCausalGeneration(BaseModel):
    text: str
